src/aqdata/aq_base.cc

```diff
@@ -89,15 +87,30 @@ void AQBase<dim>::initialize_ref_bc_index ()
 
 template <int dim>
 void AQBase<dim>::produce_angular_quad ()
-{}
+{
+  // Only provide 1D by default
+  AssertThrow (dim==1,
+               dealii::ExcMessage("Only 1D is implemented in AQBase<dim>"));
+  n_dir_ = n_azi_ / (transport_model_name_=="ep" ? 2 : 1);
+  total_angle_ = transport_model_name_=="ep" ? 4.0 : 2.0;
+  dealii::QGauss<1> mu_quad (n_azi_);
+  for ( int i=0; i<n_dir_; ++i)
+  {
+    dealii::Tensor <1,dim> omega;
+    omega[0] = mu_quad.point(i)[0]*2.0-1.0;
+    omega_i_.push_back (omega);
+    wi_.push_back (mu_quad.weight(i)*total_angle_);
+  }
+  n_total_ho_vars_ = n_dir_ * n_group_;
+}
 
 template <int dim>
 void AQBase<dim>::initialize_component_index ()
 {
   // initialize the map from (group, direction) to component indices
   int ind = 0;
-  for (unsigned int g=0; g<n_group_; ++g)
-    for (unsigned int i_dir=0; i_dir<n_dir_; ++i_dir)
+  for ( int g=0; g<n_group_; ++g)
+    for ( int i_dir=0; i_dir<n_dir_; ++i_dir)
     {
```

Comment:
Instead of changing these loop variables to `int`s, why not change `template<int dim>` to an `template<unsigned dim>`? I know we won't support negative dimensions and it's less storage and prevents having to check for out of range dimensions.

Reply:
No. we keep <int dim>, as all the dimension templating in dealii is in this fashion. We would like to keep thing consistent with deal.II.

The worst case if someone really use negative, assertion will be thrown out from deal.II functionalitites as negative dimension is prohibited.

On the other hand, if unsigned is used, we are gonna get a lot of unnecessary warnings when compiling.

Reply:
The only thing I am thinking is maybe we should change int back to `unsigned int` in for loops to erase all the warnings in compiling. This is not gonna be an performance issue for us but it does give us a clean compilation.

Reply:
Oh interesting, I didn't realize they used an `int` in dealii for dimension. That's silly, thanks!